### PR TITLE
Important Bug Fix for Super Saijack Speed Praetorians - TESTED

### PR DESCRIPTION
**Pull request by Surrealaser**
Per title.

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 1]
```diff
@@ -155,14 +155,10 @@
 	visible_message("<span class='xenowarning'>\The [src] spews forth a wide cone of acid!</span>", \
 	"<span class='xenowarning'>You spew forth a cone of acid!</span>", null, 5)
 
-	speed += 2
+	add_slowdown(2) //Getting rid of the shit code that allowed Praes to go sonic if they regained temp lost speed during upgrade
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: relying on life to fix slowdown is terrible.
assuming theres no complications with pheromones giving speed increases then you should use the caste speed as the base value to reset from.
A lot of these slowdown handling procs dont consider xeno datum stats

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 2]
```diff
@@ -120,7 +120,7 @@
 		return
 
 	if(stagger)
-		to_chat(src, "<span class='xenowarning'>Your muscles fail to respond as you try to shake up the shock!</span>")
+		to_chat(src, "<span class='xenowarning'>You're too disoriented from the shock!</span>")
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]:
```suggestion
		to_chat(src, "<span class='warning'>You're too disoriented from the shock!</span>")
```

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 3]
```diff
@@ -140,11 +140,11 @@
 	if (used_acid_spray || !check_plasma(200))
 		return
 
-	if(!do_after(src, 5, TRUE, 5, BUSY_ICON_HOSTILE))
+	if(!do_after(src, 5, TRUE, 5, BUSY_ICON_HOSTILE, TRUE))
 		return
 
 	if(stagger)
-		to_chat(src, "<span class='xenowarning'>The shock disrupts you!</span>")
+		to_chat(src, "<span class='xenowarning'>You're too disoriented from the shock!</span>")
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]:
```suggestion
		to_chat(src, "<span class='warning'>You're too disoriented from the shock!</span>")
```

File: code/modules/mob/living/carbon/xenomorph/Powers.dm (hunk 1 above)
Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/Powers.dm]: Should be fine now

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 4]
```diff
@@ -85,15 +85,15 @@
...
 		CRASH("error with caste datum")
-	xeno_caste = X
 
-	plasma_stored = xeno_caste.plasma_max
-	maxHealth = xeno_caste.max_health
+	plasma_stored = new_caste.plasma_max
+	maxHealth = new_caste.max_health
 	health = maxHealth
-	speed = xeno_caste.speed
+	speed += (new_caste.speed - old_caste.speed) //Add the difference so we don't break things if we upgrade while speed is pending modification from a temporary penalty/benefit
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm]: should just set it to the base speed of the new caste datum and then add the slowdown stacks

Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm]: Would this be better than the solution proposed? It only seems to deal with slowdown stacks as opposed to other modifiers.

**Comment by Rohesie** [comment]
This PR broke everything on the server, generating 50 MB of runtimes, making xenos unable to evolve, all be named larva (0), unable to use most abilities, immortal and other things.

It took me less than two minutes of local testing to produce these and several more:
```
runtime error: list index out of bounds
proc name: set datum (/mob/living/carbon/Xenomorph/proc/set_datum)
  source file: Xenomorph.dm,88
  usr: (src)
  src: Drone (/mob/living/carbon/Xenomorph/Drone)
  src.loc: Carpet (58,204,3) (/turf/open/floor/carpet)
  call stack:
Drone (/mob/living/carbon/Xenomorph/Drone): set datum()
Drone (/mob/living/carbon/Xenomorph/Drone): Initialize(0)
Atoms (/datum/controller/subsystem/atoms): InitAtom(Drone (/mob/living/carbon/Xenomorph/Drone), /list (/list))
Drone (/mob/living/carbon/Xenomorph/Drone): New(0)
Drone (/mob/living/carbon/Xenomorph/Drone): New(Carpet (58,204,3) (/turf/open/floor/carpet))
rohesie\'s admin datum (!local... (/datum/admins): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list))
Rohesie (/client): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list), rohesie\'s admin datum (!local... (/datum/admins))
Rohesie (/client): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list), rohesie\'s admin datum (!local... (/datum/admins))
runtime error: Cannot read null.spit_types
proc name: Initialize (/mob/living/carbon/Xenomorph/Initialize)
  source file: Xenomorph.dm,57
  usr: (src)
  src: Drone (/mob/living/carbon/Xenomorph/Drone)
  src.loc: Carpet (58,204,3) (/turf/open/floor/carpet)
  call stack:
Drone (/mob/living/carbon/Xenomorph/Drone): Initialize(0)
Atoms (/datum/controller/subsystem/atoms): InitAtom(Drone (/mob/living/carbon/Xenomorph/Drone), /list (/list))
Drone (/mob/living/carbon/Xenomorph/Drone): New(0)
Drone (/mob/living/carbon/Xenomorph/Drone): New(Carpet (58,204,3) (/turf/open/floor/carpet))
rohesie\'s admin datum (!local... (/datum/admins): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list))
Rohesie (/client): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list), rohesie\'s admin datum (!local... (/datum/admins))
Rohesie (/client): Topic("src=%5B0x210019b0%5D%3Badmin_t...", /list (/list), rohesie\'s admin datum (!local... (/datum/admins))
runtime error: Cannot read null.crit_health
proc name: update stat (/mob/living/carbon/Xenomorph/update_stat)
  source file: life.dm,64
  usr: null
  src: Drone (/mob/living/carbon/Xenomorph/Drone)
  src.loc: Carpet (58,204,3) (/turf/open/floor/carpet)
  call stack:
Drone (/mob/living/carbon/Xenomorph/Drone): update stat()
Drone (/mob/living/carbon/Xenomorph/Drone): updatehealth()
Drone (/mob/living/carbon/Xenomorph/Drone): Life(2, 7)
Drone (/mob/living/carbon/Xenomorph/Drone): Life(2, 7)
Drone (/mob/living/carbon/Xenomorph/Drone): Life(2, 7)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```

**Comment by Surrealaser** [comment]
@Rohesie tested and fixed

**Comment by spookydonut** [comment]
Obsolete

**Pull request closed without merging**